packages/hint-meta-charset-utf-8/src/hint.ts

```diff
@@ -98,15 +123,35 @@ export default class MetaCharsetUTF8Hint implements IHint {
             const isMetaElementFirstHeadContent = (/^<head[^>]*>\s*<meta/).test(headElementContent);
 
             if (!isCharsetMetaFirstHeadElement || !isMetaElementFirstHeadContent) {
-                context.report(resource, getMessage('metaElementFirstThing', context.language), { element: charsetMetaElement });
+
+                const index = html.indexOf(charsetMetaElement.outerHTML);
```

Comment:
We actually have the start and end offset from `parse5`, but haven't plumbed it through yet. I think a minor update to `HTMLElement.getLocation` to expose the start and end offset would be better than using a serialization (which may fail to find matching text at all. Best to do after the flattening PR is merged of course.

Reply:
Done, can you take a look? Had to split in a few commits